rfd/00XY-access-own-sessions.md

```diff
@@ -0,0 +1,102 @@
+---
+authors: Andrej Tokarčík (andrej@goteleport.com)
+state: draft
+---
+
+# RFD XY - Users accessing their own SSH sessions
+
+## What
+
+Notwithstanding their actual RBAC privileges, every user should be guaranteed the ability to:
+1. join the active SSH sessions they are parties to;
+2. view recordings of the SSH sessions they have participated in.
+
+Moreover, if the relevant privileges are in fact missing, the user should not be authorized to join or view any other sessions beyond the scope of (1) and (2).
+
+## Why
+
+Privileges in Teleport are structured as verb--resource pairs (e.g., `VerbList` for `KindSSHSession`) where the verb action applies to *all* resources of the kind.
+
+Unlike many other resources, SSH sessions are associated with a set of users through the notion of "party" or "participant". It is desirable to reflect this session--user relation in the authorization process so that users are always able to access their own sessions without being required to have access to all sessions/recordings indiscriminately.
+
+## Details
+
+### Joining SSH sessions involving self
+
+As it stands, every user is able to join any active session since `VerbList` and `VerbRead` for `KindSSHSession` are [granted as part of the default implicit role](https://github.com/gravitational/teleport/blob/ab57eab5c059b323e4fb50cf02c1134745a19dd1/lib/services/role.go#L79) that gets added to all role sets.
+```go
+  types.NewRule(types.KindSSHSession, RO()),
+```
+This shall be removed, with the result that no `KindSSHSession` privileges are granted implicitly.
```

Comment:
@klizhentas Removing this implicit rule wouldn't be fully backward compatible. Is that acceptable, or should we stick with the current behaviour that every user can see all active sessions unless a role explicitly denies it (as discussed in #8035)?